src/scripts/mentoring-data-migration.js

```diff
@@ -0,0 +1,1325 @@
+					WHERE ue.organization_id IS NOT NULL
+					AND ue.tenant_code IS NOT NULL
+					AND ue.tenant_code != '${defaultTenantCode}'
+					ORDER BY org_id
+				`)
+
+				console.log(`🔄 Processing ${tableName} with ${userExtByOrg.length} organizations (GROUP BY approach)`)
+
+				if (userExtByOrg.length === 0) {
+					console.log(`⚠️  No organizations found for ${tableName}, skipping updates`)
+					await transaction.commit()
+					return
+				}
+
+				let totalUpdated = 0
+
+				// Process each organization in batches
+				const orgBatchSize = 5000
+				for (let i = 0; i < userExtByOrg.length; i += orgBatchSize) {
+					const orgBatch = userExtByOrg.slice(i, i + orgBatchSize)
+
+					// Process each organization individually
+					for (const org of orgBatch) {
+						// Build SET clause based on available update columns
+						const setClauses = []
+						if (tableConfig.updateColumns.includes('tenant_code')) {
+							setClauses.push('tenant_code = ue.tenant_code')
+						}
+						if (tableConfig.updateColumns.includes('organization_code')) {
+							setClauses.push('organization_code = ue.organization_code')
+						}
+						setClauses.push('updated_at = NOW()')
+
+						const [, metadata] = await this.sequelize.query(
+							`
+							UPDATE ${tableName} 
+							SET ${setClauses.join(', ')}
+							FROM user_extensions ue
+							WHERE ${tableName}.${userIdColumn} = ue.user_id
+							AND ue.organization_id::text = '${org.org_id.replace(/'/g, "''")}'
+							AND ue.tenant_code IS NOT NULL
+							AND ue.tenant_code != '${defaultTenantCode}'
+						`,
+							{ transaction }
+						)
+
+						totalUpdated += metadata.rowCount || 0
+					}
+				}
+
+				console.log(`✅ Updated ${totalUpdated} rows in ${tableName} using GROUP BY organization_id approach`)
+				this.stats.successfulUpdates += totalUpdated
+				await transaction.commit()
+				return
+			}
+
+			// All paths above handle their own transaction commits and returns
+			// This should not be reached
+			await transaction.commit()
+		} catch (error) {
+			await transaction.rollback()
+			throw error
+		}
+	}
+
+	/**
+	 * Get lookup data for a record (legacy method for individual processing)
+	 */
+	getLookupData(record, tableConfig) {
+		const { columns } = tableConfig
+		let tenantCode = this.defaultTenantCode
+		let organizationCode = this.defaultOrgCode
+
+		// Try organization_id lookup first
+		if (columns.organization_id && record[columns.organization_id]) {
+			const orgId = record[columns.organization_id]
+			const orgData = this.orgLookupCache.get(orgId)
+			if (orgData) {
+				tenantCode = orgData.tenant_code
+				organizationCode = orgData.organization_code
+				return { tenantCode, organizationCode, source: 'organization_id' }
+			}
+		}
+
+		return { tenantCode, organizationCode, source: 'default' }
+	}
+
+	/**
+	 * Process a single table with forced updates for tenant_code and organization_code
+	 */
+	async processTable(tableConfig) {
```

Comment:
@sumanvpacewisdom this function is not used, remove this and remove related functions as well.